framework/test/unit/modules/interoperability/sidechain/commands/mainchain_registration.spec.ts

```diff
@@ -0,0 +1,426 @@
+	});
+
+	describe('execute', () => {
+		const mainchainIdAsKey = getIDAsKeyForStore(MAINCHAIN_ID);
+		const params = {
+			ownChainID: 11,
+			ownName: 'testchain',
+			mainchainValidators,
+			aggregationBits: Buffer.alloc(0),
+			signature: Buffer.alloc(0),
+		};
+		const chainAccount = {
+			name: MAINCHAIN_NAME,
+			networkID: MAINCHAIN_NETWORK_ID,
+			lastCertificate: {
+				height: 0,
+				timestamp: 0,
+				stateRoot: EMPTY_HASH,
+				validatorsHash: computeValidatorsHash(mainchainValidators, BigInt(THRESHOLD_MAINCHAIN)),
+			},
+			status: CHAIN_REGISTERED,
+		};
+		const bftParams = {
+			prevoteThreshold: BigInt(20),
+			precommitThreshold: BigInt(30),
+			certificateThreshold: BigInt(40),
+			validators: [
+				{
+					address: getRandomBytes(20),
+					bftWeight: BigInt(10),
+				},
+				{
+					address: getRandomBytes(20),
+					bftWeight: BigInt(5),
+				},
+			],
+			validatorsHash: getRandomBytes(32),
+		};
+		const blsKey1 = getRandomBytes(48);
+		const blsKey2 = getRandomBytes(48);
+		const validatorAccounts = [
+			{
+				generatorKey: getRandomBytes(48),
+				blsKey: blsKey1 < blsKey2 ? blsKey1 : blsKey2,
+			},
+			{
+				generatorKey: getRandomBytes(48),
+				blsKey: blsKey1 < blsKey2 ? blsKey2 : blsKey1,
+			},
+		];
+		const mockBFTAPI = {
+			getBFTParameters: jest.fn(),
+		};
+		const mockValidatorsAPI = {
+			getValidatorAccount: jest.fn(),
+		};
+		const mockGetStore = jest.fn();
+		const context = {
+			logger: jest.fn(),
+			eventQueue: jest.fn(),
+			networkIdentifier: Buffer.alloc(0),
+			header: {},
+			assets: {},
+			transaction,
+			params,
+			getAPIContext: jest.fn(),
+			getStore: mockGetStore,
+		} as any;
+		const chainSubstore = {
+			setWithSchema: jest.fn(),
+		};
+		const channelSubstore = {
+			setWithSchema: jest.fn(),
+		};
+		const validatorsSubstore = {
+			setWithSchema: jest.fn(),
+		};
+		const outboxRootSubstore = { setWithSchema: jest.fn() };
+		const ownChainAccountSubstore = {
+			setWithSchema: jest.fn(),
+		};
+		const sendInternal = jest.fn();
+
+		beforeEach(() => {
+			mainchainRegistrationCommand.addDependencies({
+				bftAPI: mockBFTAPI,
+				validatorsAPI: mockValidatorsAPI,
+			});
+			mainchainRegistrationCommand['getInteroperabilityStore'] = jest
+				.fn()
+				.mockReturnValue({ sendInternal });
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_DATA)
+				.mockReturnValue(chainSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHANNEL_DATA)
+				.mockReturnValue(channelSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_VALIDATORS)
+				.mockReturnValue(validatorsSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_OUTBOX_ROOT)
+				.mockReturnValue(outboxRootSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_OWN_CHAIN_DATA)
+				.mockReturnValue(ownChainAccountSubstore);
+
```

Comment:
Same as 
https://github.com/LiskHQ/lisk-sdk/pull/7116/files#r865335388

Reply:
keeping as it is for now, will add for the future tests